src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2068,6 +2068,37 @@ void ApplicationManagerImpl::ProcessServiceStatusUpdate(
       service_type, service_event, reason, app_id);
 
   rpc_service_->ManageHMICommand(notification);
+
+  if (hmi_apis::Common_ServiceEvent::REQUEST_REJECTED == service_event) {
+    StopServicesForApp(app, service_type);
+    state_ctrl_.SetRegularState(app,
+                                mobile_apis::PredefinedWindows::DEFAULT_WINDOW,
+                                mobile_apis::HMILevel::HMI_NONE,
+                                true);
+  }
+}
+
+void ApplicationManagerImpl::StopServicesForApp(
+    ApplicationSharedPtr app,
+    const hmi_apis::Common_ServiceType::eType service_type) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!app) {
+    LOG4CXX_WARN(logger_, "Received invalid app");
+    return;
+  }
+
+  switch (service_type) {
+    case hmi_apis::Common_ServiceType::VIDEO: {
+      sync_primitives::AutoLock lock(navi_service_status_lock_);
+      navi_service_status_.erase(app->app_id());
+      LOG4CXX_DEBUG(logger_,
+                    "ServiceType: " << service_type << " for appID: "
+                                    << app->app_id() << " is stopped");
+      break;
+    }
+    default:
+      break;
```

Comment:
@IGapchuk Add some logging here

Reply:
@AByzhynar Added in 3d16f70fc4034c68e2cfc2baa8fb159fb4007a83